rpcs3/rpcs3qt/pad_led_settings_dialog.cpp

```diff
@@ -0,0 +1,98 @@
+﻿#include "pad_led_settings_dialog.h"
+#include <QColorDialog>
+
+pad_led_settings_dialog::pad_led_settings_dialog(QDialog* parent)
+    : QDialog(parent), ui(new Ui::pad_led_settings_dialog)
+{
+	ui->setupUi(this);
+	setFixedSize(size());
+
+	connect(ui->b_save, SIGNAL(released()), this, SLOT(close_dialog()));
+
+	connect(this, SIGNAL(pass_led_settings(int, int, int, bool, bool, int)), parent, SLOT(apply_led_settings(int, int, int, bool, bool, int)));
+
+	connect(ui->b_apply, SIGNAL(released()), this, SLOT(pass_led_settings_signal()));
+
+	// Open QColorDialog
+	connect(ui->b_colorpicker, &QPushButton::clicked, [=]() {
+		QColor led_color(cR, cG, cB);
+		QColorDialog dlg(led_color, this);
+		dlg.setWindowTitle(tr("LED Color"));
+		if (dlg.exec() == QColorDialog::Accepted)
+		{
+			const QColor newColor = dlg.selectedColor();
+			cR = newColor.red();
+			cG = newColor.green();
+			cB = newColor.blue();
+		}
+	});
+
+	connect(ui->hs_indicator_brightness, SIGNAL(valueChanged(int)), this, SLOT(update_slider_label(int)));
+
+	connect(ui->cb_led_indicate, SIGNAL(toggled(bool)), this, SLOT(switch_groupboxes(bool)));
+}
+
+pad_led_settings_dialog::~pad_led_settings_dialog()
+{
+	delete ui;
+}
+
+// Draw an indicator for the current LED color
+void pad_led_settings_dialog::paintEvent(QPaintEvent* event)
+{
+	QPixmap pixmap(249, 39);
+	pixmap.fill(QColor("transparent"));
+	QColor led_color;
+	led_color.setRgb(cR, cG, cB);
+	QPainter painter(&pixmap);
+	painter.setRenderHint(QPainter::Antialiasing);
+	QPainterPath path;
+	path.addRoundedRect(QRectF(5, 5, 239, 29), 5, 5);
+	QPen pen(Qt::black, 1);
+	painter.setPen(pen);
+	painter.fillPath(path, led_color);
+	painter.drawPath(path);
+	ui->w_color_sample->setPixmap(pixmap);
+}
+
+// Initialize and fill the form and fields
+void pad_led_settings_dialog::init_form(int colorR, int colorG, int colorB, bool led_low_battery_blink, bool led_battery_indicator, int led_battery_indicator_brightness)
+{
+	ui->hs_indicator_brightness->setValue(led_battery_indicator_brightness);
+	ui->cb_led_blink->setChecked(led_low_battery_blink);
+	ui->cb_led_indicate->setChecked(led_battery_indicator);
+	battery_indicator_brightness = led_battery_indicator_brightness;
+	low_battery_blink = led_low_battery_blink;
+	battery_indicator = led_battery_indicator;
+	cR = colorR;
+	cG = colorG;
+	cB = colorB;
+}
+
+// Update slider's label
+void pad_led_settings_dialog::update_slider_label(int val)
+{
+	QString label = QString("%1%").arg(val);
```

Comment:
This should be slightly faster
```suggestion
	const QString label = QString::number(val) + QStringLiteral("%");
```